build-packages/scandipwa-extensibility/build-config/babel-plugin-middleware-decorator/index.js

```diff
@@ -71,9 +71,59 @@ const addSuperToConstructor = (path, types) => {
     constructor.get('body').unshiftContainer('body', superCall);
 };
 
+/**
+ * Attempt to read package.json located by p
+ * @param {string} packageJsonPath 
+ */
+const getPackageJson = (packagePath) => {
+    const packageJsonPath = path.join(packagePath, 'package.json');
+
+    try {
+        return require(packageJsonPath);
+    } catch (err) {
+        return {}
+    }
+}
+
+/**
+ * Memoization strategy is valid only for functions with 1 argument
+ * This strategy works better with 1-argument functions
+ * 
+ * @param {function} cb 
+ */
+const memoizeBySingleArgument = (cb) => {
+    const memoizedValues = {};
+
+    return (arg) => {
+        // Handle new value
+        if (!memoizedValues[arg]) {
+            const result = cb(arg);
+            memoizedValues[arg] = result;
+        }
+        
+        return memoizedValues[arg];
+    };
+}
+
+const isScandipwaPackageFile = memoizeBySingleArgument((filePath) => {
+    // Potentially, only the files inside of the src/ directory need to be transpiled
+    const src = filePath.lastIndexOf('/src/');
+
+    const packagePath = filePath.slice(0, src);
+    const packageJson = getPackageJson(packagePath);
+
+    // If package.json has `scandipwa` configurations => is a scandipwa package
+    if (packageJson.scandipwa) {
+        return true;
+    }
```

Comment:
I am not certain this is a good fix, as the `@scandipwa/form` could of very well had no `scandipwa` field. In the end of the day - it is a simple package. Maybe we should:
1. Not check for ScandiPWA
2. Strongly specify we do not support non-ScandiPWA packages. This would come with some cost though.

Reply:
I think it is a good thing to have a way to determine whether a package is a ScandiPWA package or not. 

Not only the packages of the `@scandipwa` scope should be handled as ScandiPWA packages. The day will come when the ScandiPWA ecosystem will be populated by packages contributed by community. 

How should we determine a difference between a `@namespace` comment in webpack's code (apparently it has some, it breaks without corresponding check) and in ScandiPWA code? 

I think checking for the `scandipwa` block in the `package.json` is a decent solution, if we have a requirement for a ScandiPWA package to have a type (theme, extension, maybe we'll introduce some others) the `scandipwa` block will always be there.

Reply:
Ok, then we should put a bold mark about **Application plugins will only work in the context of ScandiPWA extensions and themes!**. The `scandipwa` field should be present in `package.json` in order to be included into the candidate for a plugin.

Reply:
Add this to docs, please.

Reply:
Added here: https://app.gitbook.com/@scandipwa/s/create-scandipwa-app/extensions/extensions